packages/firestore/src/api/credentials.ts

```diff
@@ -160,10 +163,26 @@ export class FirebaseCredentialsProvider implements CredentialsProvider {
 
     this.tokenCounter = 0;
 
-    // Will fire at least once where we set this.currentUser
-    (this.app as _FirebaseApp).INTERNAL.addAuthTokenListener(
-      this.tokenListener
-    );
+    this.auth = authProvider.getImmediate(undefined, { optional: true });
+
+    if (this.auth) {
+      this.auth.addAuthTokenListener(this.tokenListener!);
+    } else {
+      // if auth is not available, invoke tokenListener once with null token
+      this.tokenListener(null);
+      authProvider.get().then(
+        auth => {
+          this.auth = auth;
+          if (this.tokenListener) {
+            // tokenListener can be removed by removeChangeListener()
+            this.auth.addAuthTokenListener(this.tokenListener!);
```

Comment:
nit: is the `!` necessary?  I would expect typescript type narrowing to make it unnecessary.

Reply:
Good catch. Removed `!`.